src/Stratis.Bitcoin/Base/TipsManager.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Base
+{
+    /// <summary>Interface that every tip provider that uses <see cref="ITipsManager"/> should implement.</summary>
+    public interface ITipProvider
```

Comment:
I suggest we rename this to TipTracker (the component that implements this is not a tip provider its rather has a need to track its current tip) WDYT